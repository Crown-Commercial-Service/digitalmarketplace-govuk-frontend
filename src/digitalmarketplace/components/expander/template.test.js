/**
 * @jest-environment jsdom
 */

const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('expander')

describe('Expander', () => {
  describe('by default', () => {
    it('matches existing snapshot', () => {
      const $ = render('expander', examples.default)
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('expander', examples.default, false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('renders the title', async () => {
      const $ = render('expander', examples.default)
      const title = $('.dm-expander__title').text().trim()

      expect(title).toEqual('Default')
    })

    it('renders the content', async () => {
      const $ = render('expander', examples.default)
      const container = $('.dm-expander__content')

      expect(container.text().trim()).toEqual('Dummy content')
    })
  })

  describe('when open on load', () => {
    it('matches existing snapshot', () => {
      const $ = render('expander', examples.expanded)
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('expander', examples.expanded, false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('sets open on load correctly', async () => {
      const $ = render('expander', examples.expanded, false, false, true)
      const container = $('.dm-expander[data-open-on-load="true"]')

      expect(container.length).toBe(1)
    })
  })

  describe('when open on load', () => {
    it('matches existing snapshot', () => {
      const $ = render('expander', examples.selected)
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('expander', examples.selected, false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('displays selected message', async () => {
      const $ = render('expander', examples.selected)
      const selectedMessage = $('.dm-expander__selected-counter').text().trim()

      expect(selectedMessage).toBe('1 selected')
    })
  })
})
