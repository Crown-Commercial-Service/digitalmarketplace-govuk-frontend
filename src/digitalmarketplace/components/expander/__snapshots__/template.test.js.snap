// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expander by default matches existing snapshot 1`] = `
"<html><head></head><body><div class=\\"dm-expander \\" data-module=\\"dm-expander\\" data-open-on-load=\\"false\\">
    <h2 class=\\"dm-expander__heading\\">
      <span class=\\"dm-expander__title js-toggle\\">Default</span>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-expander__icon dm-expander__icon--up\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m798.16 609.84l-256-256c-16.683-16.683-43.691-16.683-60.331 0l-256 256c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l225.84-225.84 225.84 225.84c16.683 16.683 43.691 16.683 60.331 0s16.683-43.691 0-60.331z\\"/></svg>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-expander__icon dm-expander__icon--down\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m225.84 414.16l256 256c16.683 16.683 43.691 16.683 60.331 0l256-256c16.683-16.683 16.683-43.691 0-60.331s-43.691-16.683-60.331 0l-225.84 225.84-225.84-225.84c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z\\"/></svg>
    </h2>
    <div class=\\"dm-expander__content js-content \\" id=\\"default\\">
      <div><p>Dummy content</p></div>
    </div>
  </div>
</body></html>"
`;

exports[`Expander when open on load matches existing snapshot 1`] = `
"<html><head></head><body><div class=\\"dm-expander \\" data-module=\\"dm-expander\\" data-open-on-load=\\"true\\">
    <h2 class=\\"dm-expander__heading\\">
      <span class=\\"dm-expander__title js-toggle\\">Expanded</span>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-expander__icon dm-expander__icon--up\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m798.16 609.84l-256-256c-16.683-16.683-43.691-16.683-60.331 0l-256 256c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l225.84-225.84 225.84 225.84c16.683 16.683 43.691 16.683 60.331 0s16.683-43.691 0-60.331z\\"/></svg>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-expander__icon dm-expander__icon--down\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m225.84 414.16l256 256c16.683 16.683 43.691 16.683 60.331 0l256-256c16.683-16.683 16.683-43.691 0-60.331s-43.691-16.683-60.331 0l-225.84 225.84-225.84-225.84c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z\\"/></svg>
    </h2>
    <div class=\\"dm-expander__content js-content dm-expander__content--visible\\" id=\\"expanded\\">
      <div><p>Dummy content</p></div>
    </div>
  </div>
</body></html>"
`;

exports[`Expander when open on load matches existing snapshot 2`] = `
"<html><head></head><body><div class=\\"dm-expander \\" data-module=\\"dm-expander\\" data-open-on-load=\\"true\\">
    <h2 class=\\"dm-expander__heading\\">
      <span class=\\"dm-expander__title js-toggle\\">Selected</span>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-expander__icon dm-expander__icon--up\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m798.16 609.84l-256-256c-16.683-16.683-43.691-16.683-60.331 0l-256 256c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l225.84-225.84 225.84 225.84c16.683 16.683 43.691 16.683 60.331 0s16.683-43.691 0-60.331z\\"/></svg>
      <svg version=\\"1.1\\" viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"dm-expander__icon dm-expander__icon--down\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"m225.84 414.16l256 256c16.683 16.683 43.691 16.683 60.331 0l256-256c16.683-16.683 16.683-43.691 0-60.331s-43.691-16.683-60.331 0l-225.84 225.84-225.84-225.84c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z\\"/></svg>
<div class=\\"dm-expander__selected-counter\\">1 selected</div>    </h2>
    <div class=\\"dm-expander__content js-content dm-expander__content--visible\\" id=\\"selected\\">
      <div><p>Dummy content</p></div>
    </div>
  </div>
</body></html>"
`;
